src/sentry/hybridcloud/migrations/0005_add_api_and_org_token_replicas.py

```diff
@@ -0,0 +1,139 @@
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0571_add_hybrid_cloud_foreign_key_to_slug_reservation"),
+        ("hybridcloud", "0004_add_cache_version"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="ApiTokenReplica",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                (
+                    "scopes",
+                    bitfield.models.BitField(
+                        [
+                            "project:read",
+                            "project:write",
+                            "project:admin",
+                            "project:releases",
+                            "team:read",
+                            "team:write",
+                            "team:admin",
+                            "event:read",
+                            "event:write",
+                            "event:admin",
+                            "org:read",
+                            "org:write",
+                            "org:admin",
+                            "member:read",
+                            "member:write",
+                            "member:admin",
+                        ],
+                        default=None,
+                    ),
+                ),
+                ("scope_list", sentry.db.models.fields.array.ArrayField(null=True)),
+                (
+                    "application_id",
+                    sentry.db.models.fields.hybrid_cloud_foreign_key.HybridCloudForeignKey(
+                        "sentry.ApiApplication", db_index=True, null=True, on_delete="CASCADE"
+                    ),
+                ),
+                (
+                    "organization_id",
+                    sentry.db.models.fields.hybrid_cloud_foreign_key.HybridCloudForeignKey(
+                        "sentry.Organization", db_index=True, null=True, on_delete="SET_NULL"
+                    ),
+                ),
+                ("application_is_active", models.BooleanField(default=False)),
+                (
+                    "user_id",
+                    sentry.db.models.fields.hybrid_cloud_foreign_key.HybridCloudForeignKey(
+                        "sentry.User", db_index=True, on_delete="CASCADE"
+                    ),
+                ),
+                (
+                    "apitoken_id",
+                    sentry.db.models.fields.hybrid_cloud_foreign_key.HybridCloudForeignKey(
+                        "sentry.ApiToken", db_index=True, on_delete="CASCADE"
+                    ),
+                ),
+                ("token", models.CharField(max_length=64)),
```

Comment:
Should this column be unique so that we can't have overlapping tokens even if replication goes sideways?

Reply:
Not quite.  I've taken the strategy of keeping this non unique so that replication, in the case of a conflict, succeeds eventually.  On the query side, I take the `.last()` item.  This greatly reduces conflict risks.

Reply:
Isn't the risk of duplicate tokens very high though? While it shouldn't happen in practice having duplicate tokens and choosing one of them could result in customers gaining access to organizations they shouldn't.

Reply:
@markstory To be clear, it's worth recalling the exact scenario where that would happen: in the world in which on control, at some point, *that token did have access*.  This will always be a problem technically -- in the face of partition, a token in the past that was valid but is not in the present valid, may still be presented.  The replication doesn't change the security story here.  Actually, by leaving this index non unique, it reduces any chance of locking up replication (because the column isn't along the shard column we can't use the delete trick we do for other unique columns), so I believe that strictly speaking it is better to take this approach.

Basically, it's worth remembering that not enforcing uniqueness is not the same as letting arbitrary tokens leak.  It's really ultimately how we handle the hand off of two valid states in different points in time, and no matter what implementation we use, eventual consistency will be an issue.

Reply:
Notably too: I can guarantee that so long as we don't treat this as a unique column, we always do converge on uniqueness with respect to the original source, it's only the potential of a hand off (circular) state that presents any complexity.

Reply:
:+1: Thanks for the reminder about potentially locking up outboxes. Relaxing the constraint is our best option to avoid that, and we need to rely on the origin table's constraint to ensure that we'll eventually settle into a state that doesn't have duplicate tokens.